File: server/src/main/kotlin/pishello/hello/cloudStorage/PhotosStorage.java
package pishello.hello.cloudStorage;

import com.google.cloud.storage.*;

import java.nio.file.Paths;
import java.util.HashMap;

public class PhotosStorage {
    
    public Bucket createNewBucket(String bucketName) throws Exception {
        Storage storage = StorageOptions.getDefaultInstance().getService();
        return storage.create(BucketInfo.of(bucketName));

    }

    public HashMap<String, String> downloadObject(String projectId, String bucketName, String objectName, String destFilePath) {
        // The ID of your GCP project
        // String projectId = "your-project-id";

        // The ID of your GCS bucket
        // String bucketName = "your-unique-bucket-name";

        // The ID of your GCS object
        // String objectName = "your-object-name";

        // The path to which the file should be downloaded
        // String destFilePath = "/local/path/to/file.txt";

        Storage storage = StorageOptions.newBuilder().setProjectId(projectId).build().getService();

        Blob blob = storage.get(BlobId.of(bucketName, objectName));
        blob.downloadTo(Paths.get(destFilePath));

        HashMap<String, String> map = new HashMap<>();
        map.put("objectName", objectName);
        map.put("bucketName", bucketName);
        map.put("Blob", blob.toString());
        return map;
    }

}

File: server/src/main/kotlin/pishello/hello/cloudStorage/PhotosStorage.kt
package pishello.hello.cloudStorage

import com.google.auth.Credentials
import com.google.auth.oauth2.GoogleCredentials
import com.google.cloud.storage.*
import org.springframework.core.io.ClassPathResource
import org.springframework.stereotype.Component
import java.nio.file.Paths


@Component
class PhotosStorage {
    val projectId = "strong-compiler-334811"
    val bucketName = "pis2021"
    val secretPath = "secret.json"

    private fun storage(): Storage {
        val credentials: Credentials = GoogleCredentials.fromStream(ClassPathResource(secretPath).inputStream)
        return StorageOptions.newBuilder()
            .setCredentials(credentials)
            .setProjectId(projectId)
            .build().service
    }

    private fun createNewBucket(bucketName: String): Bucket {
        val storage = StorageOptions.getDefaultInstance().service
        return storage.create(BucketInfo.of(bucketName))
    }

    fun write(name: String, content: String) {
        val storage = storage()
        val bucket = storage.get(bucketName)!!
        val bytes: ByteArray = content.toByteArray(Charsets.UTF_8)
        bucket.create(name, bytes)
    }

    fun read(name: String): String? {
        val storage = storage()
        val blob: Blob = storage[BlobId.of(bucketName, name)]
        return blob.getContent().toString(Charsets.UTF_8)
    }

    fun save(name: String, path: String) {
        val storage = storage()
        val blob = storage[BlobId.of(bucketName, name)]
        blob.downloadTo(Paths.get(path))
    }
}